codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/rustlang/RustTypes.kt

```diff
@@ -375,6 +375,32 @@ sealed class Attribute {
                 writer.addDependency(it.dependency)
             }
         }
+
+        companion object {
+            /**
+             * Renders a
+             * [`#[deprecated]`](https://doc.rust-lang.org/reference/attributes/diagnostics.html#the-deprecated-attribute)
+             * attribute.
+             */
+            fun deprecated(note: String? = null, since: String? = null): Custom {
+                val builder = StringBuilder()
+                builder.append("deprecated")
+
+                val note = note ?: ""
+                val since = since ?: ""
+
+                if (note.isBlank() && since.isBlank()) {
+                    // No-op. Rustc would emit a default message.
+                } else if (note.isNotBlank() && since.isBlank()) {
+                    builder.append("(note = ${note.dq()})")
+                } else if (note.isBlank() && since.isNotBlank()) {
+                    builder.append("(since = ${since.dq()})")
+                } else {
+                    builder.append("(note = ${note.dq()}, since = ${since.dq()})")
+                }
+                return Custom(builder.toString())
```

Comment:
Probably my comment was not clear in the review before. Now that you have nullable values, you can just use them and Kotlin will figure out the real type if they are not `null`. Something like this should work:

```suggestion
                if (note == null && since.null) {
                    // No-op. Rustc would emit a default message.
                } else if (note != null && since == null) {
                    builder.append("(note = ${note.dq()})")
                } else if (not == null && note != null) {
                    builder.append("(since = ${since.dq()})")
                } else {
                    builder.append("(note = ${note.dq()}, since = ${since.dq()})")
                }
                return Custom(builder.toString())
```

Reply:
Addressed. Thanks for your suggestion!